### ISSUE TITLE
not yet supported for YOLOv8 exported models

### ISSUE THREAD

**Issue by monkeycc** [ultralytics/ultralytics#952]
### Search before asking

- [X] I have searched the YOLOv8 [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/ultralytics/ultralytics/discussions) and found no similar questions.

### Question

Object detection
```
from ultralytics import YOLO

model = YOLO("yolov8n.pt")  # load a pretrained model (recommended for training)

model.train(data="VOC.yaml", epochs=100, imgsz=640)

model.export(format="onnx")
```

Results saved to AA/AA
Ultralytics YOLOv8.0.25 🚀 Python-3.8.16 torch-1.13.1 CUDA:0 (NVIDIA GeForce RTX 3060, 12050MiB)
Model summary (fused): 168 layers, 3007598 parameters, 0 gradients, 8.1 GFLOPs

PyTorch: starting from AA/AA/weights/best.pt with output shape (1, 14, 8400) (5.9 MB)

ONNX: starting export with onnx 1.13.0...
ONNX: export success ✅ 0.8s, saved as AA/AA/weights/best.onnx (11.7 MB)

Export complete (1.0s)
Results saved to /home/MM/AA/AA/weights
Predict:         yolo task=detect mode=predict model=AA/AA/weights/best.onnx -WARNING ⚠ not yet supported for YOLOv8 exported models
Validate:        yolo task=detect mode=val model=AA/AA/weights/best.onnx -WARNING ⚠ not yet supported for YOLOv8 exported models
Visualize:       https://netron.app

**Comment by JustasBart** [comment]
Hi @monkeycc ONNX exported models aren't yet officially supported via Python, but you can experiment with your model via C++ if you'd like. There's a recently added example [here](https://github.com/ultralytics/ultralytics/tree/main/examples/Yolov8_CPP_Inference).

Otherwise you'll have to either wait or to figure out how to do it by yourself :rocket:

**Comment by Laughing-q** [comment]
@monkeycc @JustasBart hi the onnx exported has been supported now! :)

**Comment by JustasBart** [comment]
@Laughing-q Oh no way! That's really cool! My apologies then seems like it's already implemented :rocket:

**Comment by glenn-jocher** [comment]
@JustasBart yes, ONNX, CoreML, OpenVINO, TensorRT and TFLite inference and validation with predict and val are now initially supported. There are some edge cases that are still not fully supported (i.e. TensorRT Segmentation inference requires a data argument), but let us know if you run into any issues.

**Comment by pderrenger** [comment]
@adrianosantospb hi there! It looks like you encountered an error related to the segmentation TensorRT (TRT) conversion. This might be due to an edge case that is still not fully supported. Could you provide more details about the specific error you encountered? This would help us in identifying the issue and providing you with a solution.
